### PR TITLE
logs for peak in-use memory

### PR DESCRIPTION
**Pull request by reusee**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #18896 

## What this PR does / why we need it:
add logs for peak in-use memory statistics.

### CODE REVIEW

**Review by daviszhen — APPROVED** [review]

**Review by zhangxu19830126 — APPROVED** [review]

File: pkg/common/malloc/peak_inuse_tracker.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+var GlobalPeakInuseTracker = NewPeakInuseTracker()
+
+func (p *PeakInuseTracker) UpdateMalloc(n uint64) {
+	for {
+		// read
+		ptr := p.ptr.Load()
+		if n <= ptr.Malloc.Value {
+			return
+		}
+		// copy
```
Comment (fengttt) [inline pkg/common/malloc/peak_inuse_tracker.go:72]: do we really need this copy?    why a few atomic int64 wont work?

**Review by qingxinhome — APPROVED** [review]

**Review by XuPeng-SH — APPROVED** [review]

File: pkg/common/malloc/peak_inuse_tracker.go (hunk 1 above)
Comment (reusee) [reply pkg/common/malloc/peak_inuse_tracker.go:72]: 并发更新 uint64 和 time.Time，要么就上锁，要么就cas。如果用cas，避免不了分配。

Comment (fengttt) [reply pkg/common/malloc/peak_inuse_tracker.go:72]: Need to allocate memory to record a stat for memory allocation.   Well.   Makes me very uncomfortable.    

Do we really need a "sync" update?

Comment (reusee) [reply pkg/common/malloc/peak_inuse_tracker.go:72]: 只有峰值被推高时，才需要复制。峰值没有推高时，就只是检查，而且不需要上锁，开销是最低的，只有 2ns。在更新少的场景，用 read-copy-update 是更合适的。还可以进一步优化，当 inuse 比记录的值大 10m 才更新，可以进一步减少开销，又不损失太大的精度。

Comment (reusee) [reply pkg/common/malloc/peak_inuse_tracker.go:77]: 加了更新阈值的优化。平均开销已经非常低了。
截屏2024-09-24 10 05 42

**Comment by reusee** [comment]
no longer required.

**Pull request closed without merging**